### DISCUSSION TITLE
Using binary transmission in PathLayer in Pydeck

### DISCUSSION THREAD

**Discussion by edwardransome** [visgl/deck.gl#7317] [Q&A]
Is it possible? The deck.gl method [here](https://deck.gl/docs/api-reference/layers/scatterplot-layer) involves passing a dict with coordinates, startIndices and length, is there a way to do this in Pydeck?
For layers with non-variable length coordinates like ScatterplotLayer the binary transmission is not an issue, but for PathLayer it seems this is currently impossible, just looking to confirm

**Comment by kylebarron** [comment]
I can't remember off the top of my head. You could try passing a `dict` of python `bytes` objects, but I don't remember if they'll get reconstructed correctly on the JS side

**Reply by edwardransome** [reply]
The Layer object only accepts pandas DataFrames as the data parameter when binary transport is set to True

```    
def _prepare_binary_data(self, data_set):
        # Binary format conversion gives a sizable speedup but requires
        # slightly stricter standards for data input
        if not is_pandas_df(data_set):
            raise BinaryTransportException("Layer data must be a `pandas.DataFrame` type")
```

**Reply by kylebarron** [reply]
Alas, maybe it's not possible today. Open to suggestions/PRs
